src/main/java/lol/server/Server.java

```diff
@@ -41,81 +34,104 @@ public void run() {
     System.out.println("Server shutted down.");
   }
 
+  public void registrationPhase() throws IOException {
+    Scanner scanner = new Scanner(System.in);
+    System.out.println("Enter the number of participants for the tourney/match: ");
+    int expectedParticipants = scanner.nextInt();
+    scanner.close();
+    while(expectedParticipants != connectedUsers.size()) {
+      waitNewPlayer();
+      System.out.println(connectedUsers.size() + "/" + expectedParticipants + " players joined.");
+    }
+    if(connectedUsers.size() > 1) { //tourney
+      System.out.println("A tourney will be started.");
+      tourneyOrganization();
+      printScores();
+    }
+    else
+      System.out.println("Not enough players to start a match.");
+  }
+
   private void waitNewPlayer() throws IOException {
     Socket socket = server.accept();
     System.out.println("New player at " + socket);
-    players.add(new Player(socket, this));
-  }
-
-  private void startGame() throws IOException {
-    championSelectionPhase();
-    spawnPhase();
-    arena.startGamePhase();
-    ui.update();
-    wait(2000);
+    connectedUsers.add(new Player(socket, this));
+    playerScores.add(0);
   }
 
-  private void wait(int timeInMS) {
-    try {
-      Thread.sleep(timeInMS); // wait time in milliseconds to control duration
-    } catch(InterruptedException e) {
-      e.printStackTrace();
+ //tourney seeded into 2 groups depending on the sign up order.
+  private void tourneyOrganization() throws IOException {
+    ArrayList<Player> group1 = new ArrayList<Player>();
+    ArrayList<Player> group2 = new ArrayList<Player>();
+    for (int i = 0; i < connectedUsers.size(); i++) {
+      if(i % 2 == 0)
+        group1.add(connectedUsers.get(i));
+      else
+        group2.add(connectedUsers.get(i));
     }
-  }
-
-  private void endOfGameMessage() {
-    for(int i = 0; i < battlefield.numberOfTeams(); ++i) {
-      Nexus nexus = battlefield.nexusOf(i);
-      if(nexus.isAlive()) {
-        System.out.println("WINNER: " + nexus);
-      }
-      else {
-        System.out.println("LOSER: " + nexus);
-      }
+    if(group2.size() < group1.size()) {
+      group2.add(null); //add dummy player.
```

Comment:
That's risky to add `null`, it could lead to `NullPointerException` of not dealt with care. Couldn't we just check the group size instead of checking the `null` in the remaining of the code?